js/src/forms/Form.js

```diff
@@ -53,7 +53,9 @@ class _Form extends React.Component {
     }
     this.setState({disabled: true, errors: {}, form_error: null})
     let r
-    const data = Object.assign({}, this.props.form_data)
+    const data = this.props.useRequestFormatter ?
+      this.props.useRequestFormatter(JSON.parse(JSON.stringify(this.props.form_data)))
```

Comment:
this is not a good way to avoid problems with mutating data!

You should fix the `useRequestFormatter` function to leave `this.props.form_data` untouched.